src/main/java/com/getbase/http/HttpClient.java

```diff
@@ -52,38 +53,53 @@ public Response request(HttpMethod method,
                             Map<String, Object> params,
                             Map<String, String> headers,
                             String body) {
-        Request.Builder builder = new Request.Builder().
-                method(method).
-                url(this.config.getBaseUrl() + url);
+        setupMDC(method, url);
 
+        try {
+            Request.Builder builder = new Request.Builder().
+                    method(method).
+                    url(this.config.getBaseUrl() + url);
 
-        if (params != null) {
-            for (Map.Entry<String, Object> q : params.entrySet()) {
-                builder.param(q.getKey(), q.getValue());
+            if (params != null) {
+                for (Map.Entry<String, Object> q : params.entrySet()) {
+                    builder.param(q.getKey(), q.getValue());
+                }
             }
-        }
 
-        applyAuthorization(builder)
-                .header("Accept", "application/json")
-                .header("User-Agent", this.config.getUserAgent())
-                .header("X-Client-Type", "api");
+            applyAuthorization(builder)
+                    .header("Accept", "application/json")
+                    .header("User-Agent", this.config.getUserAgent())
+                    .header("X-Client-Type", "api");
 
+            if (body != null) {
+                builder.header("Content-Type", "application/json").body(body);
+            }
 
-        if (body != null) {
-            builder.header("Content-Type", "application/json").body(body);
-        }
+            if (headers != null) {
+                builder.headers(headers);
+            }
 
-        if (headers != null) {
-            builder.headers(headers);
-        }
+            Response response = rawRequest(builder.build());
```

Comment:
What kind of logs do you need to track with URL and method? In jerse.HttpClient all explicit log statements track URL and method. LoggingFilter should be logging this as well.